containers/codespaces-linux/.devcontainer/library-scripts/common-debian.sh

```diff
@@ -70,6 +70,11 @@ apt-get-update-if-needed()
 
 # Run install apt-utils to avoid debconf warning then verify presence of other common developer tools and dependencies
 if [ "${PACKAGES_ALREADY_INSTALLED}" != "true" ]; then
+    # Needed for adding manpages-posix and manpages-posix-dev which are non-free packages in Debian
+    if [ "$(lsb_release -is)" = "Debian" ]; then
```

Comment:
So we can get the codename out of /etc/os-release and then use sed to do a find-replace operation. This would work on any distro but would only take affect if debian was referenced.

```bash
CODENAME="$(cat /etc/os-release | grep -oE '^VERSION_CODENAME=.+$' | cut -d'=' -f2)"
sed -i "s/deb http:\/\/deb\.debian\.org\/debian ${CODENAME} main/deb http:\/\/deb\.debian\.org\/debian ${CODENAME} main contrib non-free/" /etc/apt/sources.list
sed -i "s/deb http:\/\/deb\.debian\.org\/debian ${CODENAME}-updates main/deb http:\/\/deb\.debian\.org\/debian ${CODENAME}-updates main contrib non-free/" /etc/apt/sources.list
sed -i "s/deb http:\/\/deb\.debian\.org\/debian-security ${CODENAME}\/updates main/deb http:\/\/deb\.debian\.org\/debian-security ${CODENAME}\/updates main contrib non-free/" /etc/apt/sources.list
sed -i "s/deb http:\/\/deb\.debian\.org\/debian ${CODENAME}-backports main/deb http:\/\/deb\.debian\.org\/debian ${CODENAME}-backports main contrib non-free/" /etc/apt/sources.list
```
We also may want to do the `deb-src` equivalent of these too... and I'm sure this could be refactored into a something that processes an array of lines to replace.

In either event, this will also prevent overwriting things if someone modified sources.list (even though that's not a good idea in general) and would not add backports unless its actually in the sources.list to begin with.

That said --- looking at what we have to do here, I'm thinking we may want to have a script argument that indicates whether non-free sources should be included and only include them if this is set to true... and it probably defaults to "false". `ADD_NON_FREE_PACKAGES=${7:-"false"}`

Reply:
Sounds good. I have made all the discussed changes :)